pkg/fcDriver/nodeserver.go

```diff
@@ -0,0 +1,95 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package fc
+
+import (
+	"github.com/container-storage-interface/spec/lib/go/csi"
+	"github.com/kubernetes-csi/csi-lib-fc/fibrechannel"
+	"golang.org/x/net/context"
+)
+
+type fcNodeServer struct {
+	Driver *CSIDriver
+}
+
+var fcDisk fcDevice
+
+func (ns *fcNodeServer) NodeStageVolume(ctx context.Context, req *csi.NodeStageVolumeRequest) (*csi.NodeStageVolumeResponse, error) {
+	return &csi.NodeStageVolumeResponse{}, nil
+}
+
+func (ns *fcNodeServer) NodeUnstageVolume(ctx context.Context, req *csi.NodeUnstageVolumeRequest) (*csi.NodeUnstageVolumeResponse, error) {
+	return &csi.NodeUnstageVolumeResponse{}, nil
+}
+
+func (ns *fcNodeServer) NodePublishVolume(ctx context.Context, req *csi.NodePublishVolumeRequest) (*csi.NodePublishVolumeResponse, error) {
+	fcDevice, err := getFCInfo(req)
+
+	if err != nil {
+		return nil, err
+	}
+	fcDisk = *fcDevice
+	disk, connectError := fibrechannel.Attach(*fcDevice.connector, &fibrechannel.OSioHandler{})
```

Comment:
I think we are trying to get the "local attach" plugins such as iscsi and fc to do the attach inside the NodeStage/Unstage call to help with serialization/refcounting.  NodeStage/Unstage should only be called once per driver, while NodePublish/Unpublish is called per pod

Reply:
Sorry for the late reply, but is it just attach? Or Mount should also be in NodeStage?

Reply:
attach in nodestage, and mount in nodepublish